hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/filter/TestReclaimableRenameEntryFilter.java

```diff
@@ -0,0 +1,208 @@
+                                               SnapshotChainManager chainManager, SnapshotInfo currentSnapshotInfo,
+                                               KeyManager km, IOzoneManagerLock lock,
+                                               int numberOfPreviousSnapshotsFromChain) {
+    return new ReclaimableRenameEntryFilter(om, snapshotManager, chainManager, currentSnapshotInfo, km, lock);
+  }
+
+  List<Arguments> testReclaimableFilterArguments() {
+    List<Arguments> arguments = new ArrayList<>();
+    for (int i = 0; i < 3; i++) {
+      for (int j = 0; j < 5; j++) {
+        arguments.add(Arguments.of(i, j));
+      }
+    }
+    return arguments;
+  }
+
+  private void testReclaimableRenameEntryFilter(String volume, String bucket, int index,
+                                                String value,
+                                                Table<String, OmKeyInfo> keyTable,
+                                                Table<String, OmDirectoryInfo> dirTable,
+                                                Boolean expectedValue)
+      throws IOException {
+    List<SnapshotInfo> snapshotInfos = getLastSnapshotInfos(volume, bucket, 1, index);
+    SnapshotInfo prevSnapshotInfo = snapshotInfos.get(0);
+    OmBucketInfo bucketInfo = getOzoneManager().getBucketInfo(volume, bucket);
+    if (prevSnapshotInfo != null) {
+      ReferenceCounted<OmSnapshot> prevSnap = Optional.ofNullable(prevSnapshotInfo)
+          .map(info -> {
+            try {
+              return getOmSnapshotManager().getActiveSnapshot(volume, bucket, info.getName());
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+          }).orElse(null);
+      mockOmSnapshot(prevSnap, bucketInfo, keyTable, dirTable);
+    }
+    String key = bucketInfo.getVolumeName() + "/" + bucketInfo.getBucketName() + "/" + 1;
+    String[] keySplit = key.split("/");
+    KeyManager km = getKeyManager();
+    OMMetadataManager omMetadataManager = mock(OMMetadataManager.class);
+    when(km.getMetadataManager()).thenReturn(omMetadataManager);
+    when(omMetadataManager.splitRenameKey(eq(key))).thenReturn(keySplit);
+    assertEquals(expectedValue, getReclaimableFilter().apply(Table.newKeyValue(key, value)));
+  }
+
+  private <T> Table<String, T> getMockedTable(Map<String, T> map) throws IOException {
+    Table<String, T> table = mock(Table.class);
+    when(table.get(anyString())).thenAnswer(i -> map.get(i.getArgument(0)));
+    when(table.getIfExist(anyString())).thenAnswer(i -> map.get(i.getArgument(0)));
+    return table;
+  }
+
+  private <T> Table<String, T> getFailingMockedTable() throws IOException {
+    Table<String, T> table = mock(Table.class);
+    when(table.get(anyString())).thenThrow(new IOException());
+    when(table.getIfExist(anyString())).thenThrow(new IOException());
+    return table;
+  }
+
+  private KeyManager mockOmSnapshot(ReferenceCounted<OmSnapshot> snapshot,
```

Comment:
Returned `KeyManager` is not used anywhere? Is it supposed to return?

Reply:
we don't need it.

Reply:
removed